src/SerializeSourcemap.cc

```diff
@@ -0,0 +1,434 @@
+    }
+    else if (!element.property.empty()) {
+        return WrapPropertyMemberSourcemap(element.property); // return sos::Object
+    }
+
+    return sos::Null();                                       // return sos::Null
+}
+
+sos::Array WrapTypeSectionSourcemap(const SourceMap<mson::TypeSection>& section)
+{
+    if (!section.description.sourceMap.empty()) {
+        return WrapSourcemap(section.description);
+    }
+    else if (!section.value.sourceMap.empty()) {
+        return WrapSourcemap(section.value);
+    }
+    else if (!section.elements().collection.empty()) {
+        return WrapCollection<mson::Element>()(section.elements().collection, WrapMSONElementSourcemap);
+    }
+    return sos::Array();
+}
+
+sos::Object WrapDataStructureSourcemap(const SourceMap<DataStructure>& dataStructure)
+{
+    sos::Object dataStructureObject;
+
+    // Name
+    dataStructureObject.set(SerializeKey::Name, WrapSourcemap(dataStructure.name));
+
+    // Type Definition
+    dataStructureObject.set(SerializeKey::TypeDefinition, WrapSourcemap(dataStructure.typeDefinition));
+
+    // Type Sections
+    dataStructureObject.set(SerializeKey::Sections, 
+                            WrapCollection<mson::TypeSection>()(dataStructure.sections.collection, WrapTypeSectionSourcemap));
+
+    return dataStructureObject;
+}
+
+sos::Object WrapAssetSourcemap(const SourceMap<Asset>& asset)
+{
+    sos::Object assetObject;
+
+    // Content
+    assetObject.set(SerializeKey::Content, WrapSourcemap(asset));
+
+    return assetObject;
+}
+
+sos::Object WrapPayloadSourcemap(const SourceMap<Payload>& payload)
+{
+    sos::Object payloadObject;
+
+    // Reference
+    if (!payload.reference.sourceMap.empty()) {
+        payloadObject.set(SerializeKey::Reference, WrapSourcemap(payload.reference));
+    }
+
+    // Name
+    payloadObject.set(SerializeKey::Name, WrapSourcemap(payload.name));
+
+    // Description
+    payloadObject.set(SerializeKey::Description, WrapSourcemap(payload.description));
+
+    // Headers
+    payloadObject.set(SerializeKey::Headers, 
+                      WrapCollection<Header>()(payload.headers.collection, WrapSourcemap));
+
+    // Body
+    payloadObject.set(SerializeKey::Body, WrapSourcemap(payload.body));
+
+    // Schema
+    payloadObject.set(SerializeKey::Schema, WrapSourcemap(payload.schema));
+
+    // Content
+    sos::Array content;
+
+    /// Attributes
+    if (!payload.attributes.empty()) {
+        content.push(WrapDataStructureSourcemap(payload.attributes));
+    }
+
+    /// Asset 'bodyExample'
+    if (!payload.body.sourceMap.empty()) {
+        content.push(WrapAssetSourcemap(payload.body));
+    }
+
+    /// Asset 'bodySchema'
+    if (!payload.schema.sourceMap.empty()) {
+        content.push(WrapAssetSourcemap(payload.schema));
+    }
+
+    payloadObject.set(SerializeKey::Content, content);
+
+    return payloadObject;
+}
+
+sos::Object WrapValueSourceMap(const SourceMap<Value>& value)
```

Comment:
You named corresponding function in `SerializeAST.cc` as `WrapParameterValue` but here you are naming it `WrapValueSourcemap`. Change one or the other to make them similar.